exasol/sagemaker_extension_wrapper.py

```
@@ -0,0 +1,149 @@
+    activation SQL using the same API and writes it to the secret store. The
+    name of the key is defined in the ACTIVATION_KEY constant.
+
+    This function will eventually be shared between different extension
+    wrappers, once the language container deployment functionality is moved
+    from extensions to the script-language-container-tool repo.
+
+    Parameters:
+        conf:
+            The secret store. The store must contain the DB connection parameters
+            and the parameters of the bucket-fs service.
+        version:
+            Sagemaker Extension version.
+    """
+
+    deployer = SmeLanguageContainerDeployer.create(
+        dsn=get_external_host(conf),
+        db_user=conf.USER,
+        db_password=conf.PASSWORD,
+        bucketfs_name=conf.BUCKETFS_SERVICE,
+        bucketfs_host=conf.get("BUCKETFS_HOST_NAME", conf.EXTERNAL_HOST_NAME),
+        bucketfs_port=int(conf.BUCKETFS_PORT),
+        bucketfs_user=conf.BUCKETFS_USER,
+        bucketfs_password=conf.BUCKETFS_PASSWORD,
+        bucketfs_use_https=str_to_bool(conf, "BUCKETFS_ENCRYPTION", True),
+        bucket=conf.BUCKETFS_BUCKET,
+        path_in_bucket=PATH_IN_BUCKET,
+        use_ssl_cert_validation=str_to_bool(conf, "CERTIFICATE_VALIDATION", True),
+        ssl_trusted_ca=conf.get("TRUSTED_CA"),
+        ssl_client_certificate=conf.get("CLIENT_CERTIFICATE"),
+        ssl_private_key=conf.get("PRIVATE_KEY"),
+    )
+
+    # Install the language container.
+    deployer.download_from_github_and_run(version, False)
+
+    # Save the activation SQL in the secret store.
+    activation_sql = deployer.generate_activation_command(
+        deployer.SLC_NAME, LanguageActivationLevel.Session
+    )
+    conf.save(ACTIVATION_KEY, activation_sql)
+
+
+def deploy_scripts(conf: Secrets) -> None:
+    """
+    Deploys all the extension's scripts to the database.
+
+    Parameters:
+        conf:
+            The secret store. The store should contain the language activation
```

Should we raise an exception, in case the secret store does not contain the activation_sql?

---

Alternatively, we could enhance the SecretStore to offer a parameter or separate method that raises an exception in case of missing keys.

---

The command should be created in the previous step. Not finding the activation command is only a hypothetical possibility. It might be worth having the exception code anyway, just for completeness. But I'd rather do it in a separate ticket. In fact, this kind of ticket is coming, as Torsten wants to handle the activation slightly differently. This would be a good chance to add the exception. For now, I just want to get the whole chain of tickets merged., unless a significant issue is found.